android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java

```diff
@@ -158,6 +158,10 @@ public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
                             @Override
                             public void onCheckedChanged(
                                     CompoundButton buttonView, boolean isChecked) {
+                                if  ("noOnClickListener".equals((String) holder.assetCheck.getTag())) {
+                                    holder.assetCheck.setTag(null);
```

Comment:
I don't think I understand that part. It looks like we skip the whole block currently if we uncheck the checkbox after it's been checked? We still need to do the loop below in that case. Why we cannot just check if we unchecking and don't show the add token dialog in that case only?

Reply:
I think I see what you meant about the loop, yes the check should be put below the loop, I will correct that.

Reply:
Updated now.